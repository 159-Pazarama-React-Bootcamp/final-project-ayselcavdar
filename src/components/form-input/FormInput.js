import { useField, ErrorMessage } from 'formik';
import styles from './FormInput.module.css';
import PropTypes from 'prop-types';

export default function FormInput({ label, ...props }) {
  const { type } = props;
  const [field, meta] = useField(props);

  return (
    <div className={styles['form-container']}>
      <label className={styles['form-container--label']} htmlFor={field.name}>
        {label}
      </label>
      {type === 'textarea' ? (
        <textarea
          className={`${styles['form-control']} ${
            meta.touched && meta.error && styles['is-invalid']
          } `}
          id={field.name}
          rows="4"
          cols="50"
          {...field}
          {...props}
        />
      ) : (
        <input
          className={`${styles['form-control']} ${
            meta.touched && meta.error && styles['is-invalid']
          }`}
          id={field.name}
          {...field}
          {...props}
          autoComplete="off"
        />
      )}
      <ErrorMessage
        component="div"
        name={field.name}
        className={styles.error}
      />
    </div>
  );
}

FormInput.propTypes = {
  label: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
};
