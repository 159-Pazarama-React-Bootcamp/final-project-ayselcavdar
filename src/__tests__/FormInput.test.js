import { Formik } from 'formik';
import render from 'react-test-renderer';
import FormInput from '../components/form-input/FormInput';

describe('<FormInput /> rendering', () => {
  it('FormInput snapshots check', () => {
    const comp = render.create(
      <Formik>
        <FormInput name="test" label="test" type="text" />
      </Formik>
    );
    let tree = comp.toJSON();
    expect(tree).toMatchSnapshot();
  });
});


