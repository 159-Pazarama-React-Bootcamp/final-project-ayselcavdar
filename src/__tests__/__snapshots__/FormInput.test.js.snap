// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormInput /> rendering FormInput snapshots check 1`] = `
<div
  className="form-container "
>
  <label
    className="form-container--label"
    htmlFor="test"
  >
    test
  </label>
  <input
    autoComplete="off"
    className="form-control false"
    id="test"
    name="test"
    onBlur={[Function]}
    onChange={[Function]}
    type="text"
  />
</div>
`;

exports[`<FormInput /> rendering FormInput snapshots test 1`] = `
<div
  className="form-container "
>
  <label
    className="form-container--label"
    htmlFor="test"
  >
    test
  </label>
  <input
    autoComplete="off"
    className="form-control false"
    id="test"
    name="test"
    onBlur={[Function]}
    onChange={[Function]}
    type="text"
  />
</div>
`;

exports[`FormInput snapshots test 1`] = `
<div
  className="form-container "
>
  <label
    className="form-container--label"
    htmlFor="test"
  >
    test
  </label>
  <input
    autoComplete="off"
    className="form-control false"
    id="test"
    name="test"
    onBlur={[Function]}
    onChange={[Function]}
    type="text"
  />
</div>
`;
